bzt/resources/selenium_extras.py

```diff
@@ -16,6 +16,31 @@
     'linktext': By.LINK_TEXT
 }
 
+
+def find_element_by_shadow(shadow_loc):
+    """
+    Enables finding element using the Shadow Locator
+    :param shadow_loc: shadow locator in the string form - css locators divided by commas, e.g. 'c-comp1, c-basic, .std_btn'
+    :return: the found element otherwise NoSuchElementException is raised
+    """
+    el = None
+    css_path = [x.strip() for x in shadow_loc.split(',')]
+    for p in css_path:
+        if not el:
+            el = _get_driver().find_element_by_css_selector(p)
+        else:
+            shadow_root = el.get_property("shadowRoot")
+            if shadow_root:
+                try:
+                    el = shadow_root.find_element_by_css_selector(p)
+                except NoSuchElementException:
+                    # sometimes the element is not located under the shadowRoot so try to look for it the usual way
+                    el = el.find_element_by_css_selector(p)
```

Comment:
this is little confusing, shouldn't we only look for elements by shadow hosts?

Reply:
This is a work-around for situations where the Lightning components probably incorrectly not put the element in the shadow host... We have found this issue while testing them and this is kind of a workaround...